### PR TITLE
Annotate source code in PRs

### LINKED ISSUE #18: Add code annotations when there are test failures and/or findings-

### LINKED ISSUE #21: Findings are annotated in-line into the "files changed" source listings in PRs

### PR DESCRIPTION
**Pull request by ahtrotta**
Fixes #18 
Fixes #21 

This PR adds source code annotations to the GitHub action. For an example, see the checks on [this PR](https://github.com/land-of-apps/rails_tutorial_sample_app_7th_ed/pull/18/checks?check_run_id=15247198600).

Currently, all test failures are reported as `failures` in the check, so they will block the build, while all `new` findings are reported as `warnings` and they will not block the build.

IMO we can proceed to merge this and see how it performs in the wild. I want to follow on with a separate PR that adds a link to view the relevant AppMap in the browser. That seems like a separate chunk of work that doesn't need to block this PR. Also, we'll want to make it possible for users to configure whether any types of findings will fail the build, which also feels like a separate chunk of work.

### CODE REVIEW

File: src/Annotator.ts [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+    const lineNumber = Number(locationTokens.pop());
+    const path = locationTokens.join(':');
+    return { path, lineNumber };
+  }
+
+  public annotationFromFinding(finding: Finding): Annotation | undefined {
+    const locationInfo = finding.stack.map(this.fileAndLineNumberFromLocation, this);
+
+    let preferredLocation = locationInfo.find((info) => info && !path.isAbsolute(info.path));
```
Comment (kgilpin) [inline src/Annotator.ts]: In CI, we often see Ruby gems bundled into the _vendor_ directory, and it's also possible to see _node_modules_, so we should probably exclude any code that's in these directories, even if it's a relative path.

We should also add a property to the Action that allows the user to specify additionl directories to consider "non-local".

**Review by kgilpin — CHANGES_REQUESTED** [review]
I am seeing this fetch error in [the Analyze step](https://github.com/getappmap/analyze-action/actions/runs/5659846544/job/15334267218): 

```

/home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/@octokit/core/node_modules/@octokit/request/dist-node/index.js:57
    throw new Error(
^
Error: fetch is not set. Please pass a fetch implementation as new Octokit({ request: { fetch }}). Learn more at https://github.com/octokit/octokit.js/#fetch-missing
    at fetchWrapper (/home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/@octokit/core/node_modules/@octokit/request/dist-node/index.js:57:1)
    at request2 (/home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/@octokit/core/node_modules/@octokit/request/dist-node/index.js:172:1)
    at hook (/home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/@octokit/auth-token/dist-node/index.js:58:1)
    at /home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/@octokit/plugin-request-log/dist-node/index.js:37:1
    at /home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/before-after-hook/lib/register.js:25:1
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at Object.next (/home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/@octokit/plugin-paginate-rest/dist-node/index.js:76:1)```

**Comment by ahtrotta** [comment]
> I am seeing this fetch error in [the Analyze step](https://github.com/getappmap/analyze-action/actions/runs/5659846544/job/15334267218):
> 
> ```
> 
> /home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/@octokit/core/node_modules/@octokit/request/dist-node/index.js:57
>     throw new Error(
> ^
> Error: fetch is not set. Please pass a fetch implementation as new Octokit({ request: { fetch }}). Learn more at https://github.com/octokit/octokit.js/#fetch-missing
>     at fetchWrapper (/home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/@octokit/core/node_modules/@octokit/request/dist-node/index.js:57:1)
>     at request2 (/home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/@octokit/core/node_modules/@octokit/request/dist-node/index.js:172:1)
>     at hook (/home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/@octokit/auth-token/dist-node/index.js:58:1)
>     at /home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/@octokit/plugin-request-log/dist-node/index.js:37:1
>     at /home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/before-after-hook/lib/register.js:25:1
>     at processTicksAndRejections (node:internal/process/task_queues:96:5)
>     at Object.next (/home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/@octokit/plugin-paginate-rest/dist-node/index.js:76:1)```
> ```

This error disappears when using `getOctokit` instead of `new Octokit`, so I changed the code [here](https://github.com/getappmap/analyze-action/pull/20/commits/74b9b02c6a4e30865281c74be7e6429317776448#diff-491511c04c698dc5c1b49979afef8c4065b7fc9c4caf8d7c14f80537535c6442R81).

**Comment by kgilpin** [comment]
> > I am seeing this fetch error in [the Analyze step](https://github.com/getappmap/analyze-action/actions/runs/5659846544/job/15334267218):
> > ```
> > 
> > /home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/@octokit/core/node_modules/@octokit/request/dist-node/index.js:57
> >     throw new Error(
> > ^
> > Error: fetch is not set. Please pass a fetch implementation as new Octokit({ request: { fetch }}). Learn more at https://github.com/octokit/octokit.js/#fetch-missing
> >     at fetchWrapper (/home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/@octokit/core/node_modules/@octokit/request/dist-node/index.js:57:1)
> >     at request2 (/home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/@octokit/core/node_modules/@octokit/request/dist-node/index.js:172:1)
> >     at hook (/home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/@octokit/auth-token/dist-node/index.js:58:1)
> >     at /home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/@octokit/plugin-request-log/dist-node/index.js:37:1
> >     at /home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/before-after-hook/lib/register.js:25:1
> >     at processTicksAndRejections (node:internal/process/task_queues:96:5)
> >     at Object.next (/home/runner/work/analyze-action/analyze-action/dist/webpack:/preflight-appmap-action/node_modules/@octokit/plugin-paginate-rest/dist-node/index.js:76:1)```
> > ```
> 
> This error disappears when using `getOctokit` instead of `new Octokit`, so I changed the code [here](https://github.com/getappmap/analyze-action/pull/20/commits/74b9b02c6a4e30865281c74be7e6429317776448#diff-491511c04c698dc5c1b49979afef8c4065b7fc9c4caf8d7c14f80537535c6442R81).

Weird. Good job finding the fix.

**Review by kgilpin — APPROVED** [review]

**Pull request merged**